### ISSUE TITLE
5.0.1 iOS - Cancel does not close email modal/menu

### ISSUE THREAD

**Issue by cristianoccazinsp** [chirag04/react-native-mail#157]
The email modal does not close after sending or pressing the cancel button on iOS (13.6).

Mailer 5.0.1
RN: 0.62.2

**Comment by Relax594** [comment]
which modal? No issues here on iOS 13.6.1.

**Comment by cristianoccazinsp** [comment]
The email composing interface (overlay?) was not disappearing after hitting send/cancel.  However, looks like this was a usage issue and not really a bug.

Basically, I was unmounting the component (modal) right after prompting for the email composer, which causes issues in the UI hierarchy.  You may disregard this.

**Comment by andrzejbe** [comment]
@Relax594 I have the same issue as @cristianoccazinsp but I actually think this could be a bug.

I have a RN app with context menu on various screens - the menu itself is built with react-native-modal which is a wrapper around RN Modal component.

Some menu items in that context menu triggers displaying a different modals - and because a limitation in how RN handles modals, we handle it sequentially  - when user taps an option that should open different modal we first close the (original) context menu modal, wait until its completely hidden (I use onModalHide prop from react-native-modal for this) and once its unmounted, only then we mount/launch the next modal.

The same pattern/behaviour worked for me when I used to launch Mail Composer modal from react-native-mail - but ever since we upgraded to RN 0.64 it stopped working on iOS - now, when from within the context menu (a Modal) I tap on "send email" option and:

- hide and wait for unmount of the modal where the event was triggered
- only then trigger Mailer.mail() - next modal

... the Composer window does not start and instead I see the following error:

`Warning: Attempt to present <MFMailComposeViewController: 0x13e150c00> on <UIViewController: 0x13d6bc8e0> whose view is not in the window hierarchy!`

It seems like the Mailer modal is being attached to the UI component (modal) from which the source event originated - and in case of RN modals, as the original modal at this point is already unmounted when its finally time to launch Composer view, it isn't correctly attached to a mounted View in underlying windows hierarchy [I presume].

Of course, a work-around could be that I DONT unmount the original modal (which was @cristianoccazinsp solution] - but this will mean that once the user sends the email they go back to a view with still-opened context menu - which is not ideal.

Is there any chance Mailer Composer View is attached NOT to the View where the original event was triggered (in my case - a Modal) - but to some root / parent App view instead? [so that it launches correctly even when a modal where is was trigged is being unmounted fro the view] ..?
